patterns/cell-based-eks/1.cell1/main.tf

```diff
@@ -0,0 +1,349 @@
+provider "aws" {
+  region = local.region
+}
+
+# Required for public ECR where Karpenter artifacts are hosted
+provider "aws" {
+  region = "us-east-1"
+  alias  = "virginia"
+}
+
+data "terraform_remote_state" "vpc" {
+  backend = "local"
+
+  config = {
+    path = "${path.module}/../0.vpc/terraform.tfstate"
+  }
+}
+
+data "aws_ecrpublic_authorization_token" "token" {
+  provider = aws.virginia
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+    }
+  }
+}
+
+provider "kubectl" {
```

Comment:
we are actively moving away from this provider https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1675

You can see the start of this in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1819 which also includes updates for Karpenter 0.32

Reply:
Removed the use of kubectl provider and added instructions in README.md to install the karpenter and sample app.